src/main/java/seedu/address/model/person/LessonTime.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.EnumUtil.inEnum;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+/**
+ * Represents a LessonTime in the address book.
+ * Guarantees: immutable; value is valid as declared in {@link #isValidLessonTime(String)}
+ */
+public class LessonTime {
+    enum Days {
+        MON, TUE, WED, THU, FRI, SAT, SUN
+    }
+
+    public static final String MESSAGE_CONSTRAINTS = "Lesson time should be in format DAY-STARTTIME-ENDTIME, where: "
+            + "\n" + "DAY is in " + Arrays.toString(Days.values()) + ";\n"
+            + "STARTTIME and ENDTIME are in the format HH:MM;\n"
+            + "ENDTIME is later than STARTTIME.";
+    private static final DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm");
+    private final String lessonDay;
+    private final LocalTime lessonStart;
+    private final LocalTime lessonEnd;
+
+    /**
+     * Constructs a {@code LessonTime}
+     * @param lessonTime A valid lesson time.
+     */
+    public LessonTime(String lessonTime) {
+        requireNonNull(lessonTime);
```

Comment:
maybe can have parsers to aid in this like parser to handle this details? then have it as LessonTime(Day lessonDay, LocalTime lessonstart, localtime lessonday)?
cos the other fields have parsers. i hope they all do hahah idk

Reply:
yup theres alr parsers, its in ParserUtil

Reply:
yup there's already parsers, they're in ParserUtil